File: src/Api/Model/DB/CNAE/CnaeSecao.cs
using Realms;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CriadorBaseDados.Model.DB.CNAE
{
    /// <summary>
    /// Seção -> Divisão -> Grupo -> Classe -> Subclasse -> Atividade Econômica
    /// </summary>
    public class CnaeSecao : RealmObject
    {
        [PrimaryKey]
        public string ID { get; set; }
        public string Descricao { get; set; }
        //public string Observacoes { get; set; }
        [Backlink(nameof(CnaeDivisao.Secao))]
        public IQueryable<CnaeDivisao> Divisoes { get; }
    }
}

File: src/Mobile/b2b/lib/model/core/cnae/cnae_secao.dart
import 'cnae_divisao.dart';

/// Seção -> Divisão -> Grupo -> Classe -> Subclasse -> Atividade Econômica
/// </summary>
class CnaeSecao {
  final String id;
  final String descricao;
  final List<CnaeDivisao> divisoes;

  CnaeSecao(this.id, this.descricao, this.divisoes);

  CnaeSecao.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        descricao = json['descricao'],
        divisoes = json['divisoes'] ?? List.empty(growable: true);

  Map<String, dynamic> toMap() => <String, dynamic>{
        'id': id,
        'descricao': descricao,
        //'divisoes': divisoes,
      };
}
